internal/auth/oidc/repository_auth_method.go

```diff
@@ -0,0 +1,147 @@
+func (r *Repository) UpdateAuthMethod(ctx context.Context, m *AuthMethod, version uint32, fieldMaskPaths []string, _ ...Option) (*AuthMethod, error) {
+	panic("to-do")
+}
+
+// TestAuthMethod will test/validate the provided AuthMethod.
+//
+// It will verify that all required fields for a working AuthMethod have values.
+//
+// If the AuthMethod contains a DiscoveryUrl for an OIDC provider, TestAuthMethod
+// retrieves the OpenID Configuration document. The values in the AuthMethod
+// (and associated data) are validated with the retrieved document. The issuer and
+// id token signing algorithm in the configuration are validated with the
+// retrieved document. TestAuthMethod also verifies the authorization, token,
+// and user_info endpoints by connecting to each and uses any certificates in the
+// configuration as trust anchors to confirm connectivity.
+//
+// No options are currently supported.
+func (r *Repository) TestAuthMethod(ctx context.Context, m *AuthMethod, opt ...Option) error {
+	panic("to-do")
+}
+
+// MakeInactive will transision an OIDC auth method from either the
+// ActivePrivateState or the ActivePublicState into the temporary StoppingState
+// and then, after a small amount of time, to the InactiveState.
+func (r *Repository) MakeInactive(ctx context.Context, authMethodId string, _ ...Option) error {
+	panic("to-do")
+}
+
+// MakePrivate will transision an OIDC auth method from either the
+// InactiveState or the ActivePublicState into the temporary StoppingState
+// and then, after a small amount of time, to the ActivePrivateState.  If
+// transitioning from the InactiveState, the transition will only succeed if
+// the oidc.TestAuthMethod(...) succeeds. No options are currently supported.
+func (r *Repository) MakePrivate(ctx context.Context, authMethodId string, opt ...Option) error {
+	panic("to-do")
+}
+
+// MakePublic will transision an OIDC auth method from either the
+// InactiveState or the ActivePrivateState into the temporary StoppingState
+// and then, after a small amount of time, to the ActivePublicState.  If
+// transitioning from the InactiveState, the transition will only succeed if
+// the oidc.TestAuthMethod(...) succeeds. No options are currently supported.
+func (r *Repository) MakePublic(ctx context.Context, authMethodId string, opt ...Option) error {
+	panic("to-do")
+}
+
+// StartAuth accepts a request to start an OIDC authentication/authorization
+// attempt containing the client_type, the client_version, and a map of client
+// round-tripper k/v pairs.  It returns two URLs.  authUrl is an OIDC
+// authorization request URL.  tokenUrl is the URL the client can use to retrieve
+// the results of the user's OIDC authentication attempt.  No options are
+// currently supported.
+func (r *Repository) StartAuth(ctx context.Context, authMethodId string, clientType, clientVersion string, clientRoundTripKVs map[string]string) (authUrl *url.URL, tokenUrl *url.URL, e error) {
```

Comment:
I think the `clientType`, `clientVersion`, and `clientRoundTripKVs` parameters should be combined into a single struct since they are related. Same thing with the `authUrl` and `tokenUrl` return values since they are tightly related.

Reply:
Agreed that they are related, but I'm not sure they should be combined into a struct unless there's some ambiguity across the existing function parameters.  Embedding them within a struct doesn't make the function signature more readable.  I might argue it makes it less readable, especially in the godocs.   All this applies to the return values as well.  There's no ambiguity about the purpose/intent of authUrl and tokenUrl.  Happy and interested to discuss more about this.